File: tests/src/test/java/com/github/klee0kai/stone/test/interfaceprovide/EarthWayTest.java
package com.github.klee0kai.stone.test.interfaceprovide;

import com.github.klee0kai.stone.Stone;
import com.github.klee0kai.test.di.earthmirror.EarthComponent;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertNotNull;

public class EarthWayTest {

    @Test
    public void ChristopherColumbusTest() {
        //When
        EarthComponent di = Stone.createComponent(EarthComponent.class);

        //Then
        assertNotNull(di.west().mountainImp());
        assertNotNull(di.west().riverImpl());
    }
}

File: tests/src/test/java/com/github/klee0kai/stone/test/interfaceprovide/EarthWayTest.kt
package com.github.klee0kai.stone.test.interfaceprovide

import com.github.klee0kai.test.di.earthmirror.EarthComponentStoneComponent
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.assertNotNull

class EarthWayTest {

    @Test
    fun christopherColumbusTest() {
        //When
        val di = EarthComponentStoneComponent()

        //Then
        assertNotNull(di.west().mountainImp())
        assertNotNull(di.west().riverImpl())
    }

}
